stl/inc/ranges

```diff
@@ -6837,6 +6837,602 @@ namespace ranges {
+            using _Common_size_type          = common_type_t<range_size_t<_ViewTypes>...>;
+            using _Common_unsigned_size_type = _Make_unsigned_like_t<_Common_size_type>;
+
+            auto _Smallest_view_size_value = numeric_limits<_Common_unsigned_size_type>::max();
+
+            _Zip::_Tuple_for_each(
+                [&_Smallest_view_size_value]<typename _View>(_View& _Curr_view) {
+                    const auto _Curr_view_size_value =
+                        static_cast<_Common_unsigned_size_type>(_RANGES size(_Curr_view));
+
+                    if (_Curr_view_size_value < _Smallest_view_size_value)
+                        _Smallest_view_size_value = _Curr_view_size_value;
+                },
+                _M_Views);
+
+            return _Smallest_view_size_value;
+        }
+
+        _NODISCARD constexpr auto size() const
+            noexcept(noexcept(_Zip::_Tuple_for_each(_RANGES size, _M_Views))) // strengthened
+            requires(sized_range<const _ViewTypes>&&...) {
+            using _Common_size_type          = common_type_t<range_size_t<_ViewTypes>...>;
+            using _Common_unsigned_size_type = _Make_unsigned_like_t<_Common_size_type>;
+
+            auto _Smallest_view_size_value = numeric_limits<_Common_unsigned_size_type>::max();
+
+            _Zip::_Tuple_for_each(
+                [&_Smallest_view_size_value]<typename _View>(const _View& _Curr_view) {
+                    const auto _Curr_view_size_value =
+                        static_cast<_Common_unsigned_size_type>(_RANGES size(_Curr_view));
+
+                    if (_Curr_view_size_value < _Smallest_view_size_value)
+                        _Smallest_view_size_value = _Curr_view_size_value;
+                },
+                _M_Views);
+
+            return _Smallest_view_size_value;
+        }
+
+    private:
+        _NODISCARD constexpr auto _Get_signed_size() noexcept(noexcept(size())) {
+            const auto _Possibly_unsigned_size = size();
+
+            if constexpr (is_unsigned_v<decltype(_Possibly_unsigned_size)>) {
+                return static_cast<_Make_signed_like_t<decltype(_Possibly_unsigned_size)>>(_Possibly_unsigned_size);
+            } else {
+                return _Possibly_unsigned_size;
+            }
+        }
+
+        _NODISCARD constexpr auto _Get_signed_size() const noexcept(noexcept(size())) {
+            const auto _Possibly_unsigned_size = size();
+
+            if constexpr (is_unsigned_v<decltype(_Possibly_unsigned_size)>) {
+                return static_cast<_Make_signed_like_t<decltype(_Possibly_unsigned_size)>>(_Possibly_unsigned_size);
+            } else {
+                return _Possibly_unsigned_size;
+            }
+        }
+
+        template <bool _IteratorConst>
+        _NODISCARD static constexpr const auto& _Get_iterator_tuple(const _Iterator<_IteratorConst>& _Itr) noexcept {
+            return _Itr._M_Current;
+        }
+
+    private:
+        tuple<_ViewTypes...> _M_Views;

```

Comment:
```suggestion
        /* [[no_unique_address]] */ tuple<_ViewTypes...> _M_Views;
```
We usually name underlying view[s] as `_Range[s]` (and we don't use `_M_` prefix) and add `/* [[no_unique_address]] */` comment.
Also, I think it should be defined at the beginning of the class.

Reply:
Right, that makes sense. I'll need to rename and move the members for all of the implemented classes, then, but for consistency, it should be done. I'll make a separate commit for this.

Reply:
If I may ask, though, why is the `/* [[no_unique_address] */` comment needed? They are objects, so by that definition in the C++ standard, they should indeed have a unique address, right? If it so happens that this is not the case for a specific range type, then why even use a comment, rather than just actually declaring the member as `[[no_unique_address]]`?

Perhaps I am misunderstanding something here, but I'm hesitant to add this until I know why exactly it is needed. I'll go ahead and add the other suggestions, though.

Reply:
`[[no_unique_address]]` was implemented in MSVC by not doing anything at all. These comments show where we should add `[[no_unique_address]]` when we're able to do an ABI breaking release.

Reply:
Alright, that makes sense. I'll add such a comment there. In retrospect, it makes sense to add the attribute for the `std::tuple` of ranges. After all, if a range type `R` was able to make use of `[[no_unique_address]]`, then so should `std::tuple<R>`.

Thanks for explaining what the comments were for, though! It's kind of sad to head that that's how things are in the MSVC, but I'll just keep using the empty base optimization method in my own code.